File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
using namespace std;
class Stack{
private:
    int n;
    int stack[10000];
    int size;
public:
    Stack(){
        this->size = 0;
    }
    void push(int data){
        stack[size] = data;
        size+=1;
    }
    int pop(void){
        if(size == 0)
            return -1;
        size-=1;
        return stack[size];
    }
    int put_size(void){
        return size;
    }
    bool empty(void){
        return (size==0);
    }
    int top(void){
        if(size == 0)
            return -1;
        return stack[size-1];
    }
};

int main(void) {
    int n;
    cin >> n;
    string command;
    Stack stack;
    
    for(int i = 0;i<n;i++){
        cin >> command;
        if(command == "push"){
            int data;
            cin >> data;
            stack.push(data);
        }
        else if(command == "pop"){
            cout << stack.pop() << endl;
        }
        else if(command == "size"){
            cout << stack.put_size() << endl;
        }
        else if(command == "empty"){
            cout << stack.empty() << endl;
        }
        else if(command == "top"){
            cout << stack.top() << endl;
        }
        else{
            cout << "Wrong Command!" << endl;
        }
        
    }
    return 0;
}

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main{
    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        Stack<Integer> stack = new Stack<>();
        int N  = Integer.parseInt(bf.readLine());
        for(int i = 0;i<N;i++){
            String userInput = bf.readLine();
            String[] tokens = userInput.split(" ");
            boolean checkEmpty = stack.isEmpty();
            switch (tokens[0]){
                case "push":
                    stack.push(Integer.parseInt(tokens[1]));
                    break;
                case "pop":
                    if(checkEmpty){
                        System.out.println(-1);
                    }else{
                        System.out.println(stack.pop());
                    }
                    break;
                case "size":
                    System.out.println(stack.size());
                    break;
                case "empty":
                    if(checkEmpty){
                        System.out.println(1);
                    }else{
                        System.out.println(0);
                    }
                    break;
                case "top":
                    if(checkEmpty){
                        System.out.println(-1);
                    }else{
                        System.out.println(stack.peek());
                    }
                    break;
            }

        }

    }
}